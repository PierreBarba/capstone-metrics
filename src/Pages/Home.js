import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { getApiCountries } from '../Redux/CountriesReducer';
import MapList from '../Components/MapList';
import store from '../Redux/ConfigureStore';
import microfon from '../images/microfon.png';
import config from '../images/gear.png';
import europe from '../images/europe.png';

let flag = false;
const Home = () => {
  const [data, setData] = useState('');
  const countries = useSelector((state) => state.Countries);

  useEffect(() => {
    if (!flag) {
      store.dispatch(getApiCountries());
      flag = true;
    }
  }, []);

  const onChange = (event) => {
    setData(event.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
  };

  return (
    <>
      <div className="navBar">
        <h2 className="europeNav"> Europe </h2>
        <img src={config} alt="config" className="iconSet" />
      </div>
      <div className="home-page">
        <div className="header" id="flex">
          <div className="containerImgEuro">
            <img src={europe} alt="config" className="imgEurope" />
          </div>
          <div className="containerForm">
            <h1 className="main-title">EUROPE&apos;S COUNTRY</h1>
            <div className="form-section" id="flex">
              <form className="input-search">
                <input
                  onSubmit={onSubmit}
                  className="find-item"
                  type="search"
                  onChange={onChange}
                  placeholder="your europe country"
                />
              </form>
              <img src={microfon} alt="microfon" className="icon" />
            </div>
          </div>
        </div>
        <div className="navCountries">
          <h2 className="h2countries"> Countries </h2>
        </div>
        <MapList
          CountriesData={countries.filter((country) => (
            data.toLowerCase() === '' ? country : country.name.toLowerCase().includes(data)
          ))}
        />
        <div className="up">
          <a href="navBar" className="up-arrow">
            <button className="buttonArrow" type="button">&#128316;</button>
          </a>
        </div>
      </div>
    </>
  );
};

export default Home;
