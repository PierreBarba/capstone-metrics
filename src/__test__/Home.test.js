import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Pages/Home';
import '@testing-library/jest-dom';
import store from '../Redux/ConfigureStore';

describe('correct deployment of the home page', () => {
  test('Country api components matches snapshot', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});