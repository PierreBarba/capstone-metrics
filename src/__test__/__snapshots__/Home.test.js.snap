// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct deployment of the home page Country api components matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navBar"
      >
        <h2
          class="europeNav"
        >
           Europe 
        </h2>
        <img
          alt="config"
          class="iconSet"
          src="gear.png"
        />
      </div>
      <div
        class="home-page"
      >
        <div
          class="header"
          id="flex"
        >
          <div
            class="containerImgEuro"
          >
            <img
              alt="config"
              class="imgEurope"
              src="europe.png"
            />
          </div>
          <div
            class="containerForm"
          >
            <h1
              class="main-title"
            >
              EUROPE'S COUNTRY
            </h1>
            <div
              class="form-section"
              id="flex"
            >
              <form
                class="input-search"
              >
                <input
                  class="find-item"
                  placeholder="your europe country"
                  type="search"
                />
              </form>
              <img
                alt="microfon"
                class="icon"
                src="microfon.png"
              />
            </div>
          </div>
        </div>
        <div
          class="navCountries"
        >
          <h2
            class="h2countries"
          >
             Countries 
          </h2>
        </div>
        <div
          class="countriesgroup"
        />
        <div
          class="up"
        >
          <a
            class="up-arrow"
            href="navBar"
          >
            <button
              class="buttonArrow"
              type="button"
            >
              ðŸ”¼
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navBar"
    >
      <h2
        class="europeNav"
      >
         Europe 
      </h2>
      <img
        alt="config"
        class="iconSet"
        src="gear.png"
      />
    </div>
    <div
      class="home-page"
    >
      <div
        class="header"
        id="flex"
      >
        <div
          class="containerImgEuro"
        >
          <img
            alt="config"
            class="imgEurope"
            src="europe.png"
          />
        </div>
        <div
          class="containerForm"
        >
          <h1
            class="main-title"
          >
            EUROPE'S COUNTRY
          </h1>
          <div
            class="form-section"
            id="flex"
          >
            <form
              class="input-search"
            >
              <input
                class="find-item"
                placeholder="your europe country"
                type="search"
              />
            </form>
            <img
              alt="microfon"
              class="icon"
              src="microfon.png"
            />
          </div>
        </div>
      </div>
      <div
        class="navCountries"
      >
        <h2
          class="h2countries"
        >
           Countries 
        </h2>
      </div>
      <div
        class="countriesgroup"
      />
      <div
        class="up"
      >
        <a
          class="up-arrow"
          href="navBar"
        >
          <button
            class="buttonArrow"
            type="button"
          >
            ðŸ”¼
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
